src/models/simple_metabolism_model.py

```diff
@@ -135,22 +130,21 @@ def get_iob_from_sbr(self, sbr_actual):
         num_hours_post_t0 = 8
         minutes_per_pump_pulse = 5
 
-        # E.g. 1 pulse every 5 minutes: pulses/hr = 60 min/hr / 5 min/pulse
+        # E.g. 1 pulse every 5 minutes: 12 pulses/hr = 60 min/hr / 5 min/pulse
         num_basal_pulses_per_hour = int(MINUTES_PER_HOUR / minutes_per_pump_pulse)
 
         # TODO: CS - this is unrealistic pump behavior since pumps deliver in increments of U/pulse
         basal_amount_per_5min = (
             sbr_actual / num_basal_pulses_per_hour
         )  # U/pulse = U/hr / pulse/hr
 
-        t = get_timeseries(num_hours=num_hours_pre_t0, five_min=False)
-        t_5min = get_timeseries(num_hours=num_hours_pre_t0, five_min=True)
-        i_t, iob_t = self.insulin_model.run(t, insulin_amount=basal_amount_per_5min)
-        iob_t5min = iob_t[t_5min]
+        t, delta_bg_t, bg_t, iob_t = self.insulin_model.run(num_hours_pre_t0,
+                                                            insulin_amount=basal_amount_per_5min,
+                                                            five_min=True)
 
         # Step 2: Add hours post t0 to simulation
         iob_with_zeros = np.append(
-            iob_t5min, np.zeros(num_hours_post_t0 * num_basal_pulses_per_hour)
+            iob_t, np.zeros(num_hours_post_t0 * num_basal_pulses_per_hour)
```

Comment:
This comment is re: line 161. Yes, I believe this is necessary because the roll function will take the data that was at the end of the array and will shift it to the beginning (e.g. array -> [1, 2, 3] with a roll of 1 becomes [3, 1, 2 ] and the behavior that we really want is [0, 1, 2].

Reply:
Oh I see, I thought the tri() function was serving a different purpose and that my numpy version might not be revealing it. I think actually in this case it isn't necessary since the roll never wraps any non-zero values in any of the columns. It'd be good to conclude on this, because if that's the case we could speed this function a bit.